readthedocs/doc_builder/python_environments.py

```diff
@@ -63,13 +68,27 @@ def delete_existing_venv_dir(self):
             ))
             shutil.rmtree(venv_dir)
 
-    def install_package(self):
-        if (self.config.python.install_with_pip or
-                getattr(settings, 'USE_PIP_INSTALL', False)):
```

Comment:
I almost forgot I removed this (it wasn't correctly setup there anyway). We have the config file v1 and v2 to install using pip, not sure if this is used, we can remove this (there are some docs that need to be updated too)

Reply:
We have https://docs.readthedocs.io/en/latest/settings.html#use-pip-install, but the logic wasn't correct, I mean, I think this was used when we didn't have a config object, and this will change the behavior of the config from the web `Install project using setup` to use pip instead. So, this will invert that setting, but now we have the two settings, install with pip or setup, so, we can't _guess_ or rely on `install_with_setup` or whatever to invert the meaning.

I think the logic before the refactor to support v2 was correct and was inverting the behavior if the user has the config from the web marked as `install with setup` that should work.

This looks like something not so useful now, and if someone was relying on it it was only on a custom installation. If you think this is still useful and the logic should be fixed, I can check for this flag when setting the defaults (here we can distingue the configs from the web)

Reply:
I don't think we need the setting, we're not using this in production.

Regarding the database setting, I think we should be using this. As far as I can tell we aren't using that in this PR?

Reply:
We are passig the value as default for v1 here https://github.com/rtfd/readthedocs.org/blob/4031b9078efba9948301e89c948a236671086d26/readthedocs/doc_builder/config.py#L49-L49

Ans v1 is setting that value here 

https://github.com/rtfd/readthedocs.org/blob/4031b9078efba9948301e89c948a236671086d26/readthedocs/config/config.py#L423-L432